ruby/lib/rspec/queue.rb

```diff
@@ -142,8 +176,67 @@ def run(reporter)
       end
     end
 
+    class ReportRunner
+      include RunnerHelpers
+      include CI::Queue::OutputHelpers
+
+      def call(options, stdout, stderr)
+        setup(options, stdout, stderr)
+
+        queue = CI::Queue.from_uri(queue_url, RSpec::Queue.config)
+
+        supervisor = begin
+          queue.supervisor
+        rescue NotImplementedError => error
+          abort! error.message
+        end
+
+        step("Waiting for workers to complete")
+
+        unless supervisor.wait_for_workers
+          unless supervisor.queue_initialized?
+            abort! "No master was elected. Did all workers crash?"
+          end
+
+          unless supervisor.exhausted?
+            abort! "#{supervisor.size} tests weren't run."
+          end
+        end
+
+        # TODO: better reporting
+        errors = supervisor.build.error_reports.sort_by(&:first).map(&:last)
```

Comment:
Curious as to what the structure of  `error_reports` is that requires `.sort_by(&:first).map(&:last)`.

Reply:
It's a `Hash[Example#id => String]`.

`Example#id` looks like: `./spec/foo_spec[1:1]`

The sort is so that errors coming from the same file are grouped together.

Reply:
Thanks for the explanation 👍